### PR TITLE
Oauth automatic redirect

### PR DESCRIPTION
**Pull request by fenix-hub**
Related to #203 
This PR implements the possibility to automatically login/signup using OAuth2 Authentication method

Two new functions are implemented
1) `get_google_auth_redirect(redirect_uri : String, listen_to_port : int)` to set a custom `redirect_uri` where Google will send the token required to be exchanged for a Firebase one
2) `get_google_auth_localhost(port : int)` to set the loopback ip as the `redirect_uri` to get the token back automatically

Both functions listen to the port `49152` by default.

While the first function needs to be tested using different URI other than the loopback one, the second function has been tested.

Once `get_google_auth_localhost()` is called, the browser will automatically open to asking the user to use a Google account o authenticate. Then the user will be redirected to the redirect_uri chosen (which is `localhost:49152` for this function) and Godot will automatically get the token. Once the token is retrieved, the plugin will automatically exchange the token a make a sign up request.

**NOTE:** `get_google_auth()` is now renamed to `get_google_auth_manual()`

### CODE REVIEW

File: addons/godot-firebase/auth/auth.gd [hunk 1]
```diff
@@ -435,4 +479,10 @@ func get_clean_keys(auth_result : Dictionary) -> Dictionary:
...
     return cleaned
 
-
+func _url_decode(url : String) -> String:
+    var decoded_url : String = url.replacen("%21", "!").replacen("%23", "#").replacen("%24", "$") \
+    .replacen("%26", "&").replacen("%27", "'").replacen("%28", "(").replacen("%29", ")") \
+    .replacen("%2A", "*").replacen("%2B", "+").replacen("%2C", ",").replacen("%2F", "/") \
+    .replacen("%3A", ":").replacen("%3B", ";").replacen("%3D", "=").replacen("%3F", "?") \
+    .replacen("%40", "@").replacen("%5B", "[").replacen("%5D", "]")
+    return decoded_url
```
Comment (WolfgangSenff) [inline addons/godot-firebase/auth/auth.gd:488]: Doesn't https://docs.godotengine.org/en/stable/classes/class_string.html#class-string-method-percent-decode handle this, or does it do something you don't want? Or not do a complete job? I guess this is all weird characters, hm.

File: addons/godot-firebase/containers/circular_container/circular_container.gd [hunk 2]
```diff
@@ -0,0 +1,44 @@
+tool
```
Comment (WolfgangSenff) [inline addons/godot-firebase/containers/circular_container/circular_container.gd:1]: Didn't we remove these files from here and put them all in GodotFirebase-UI?

**Review by WolfgangSenff — CHANGES_REQUESTED** [review]
Aside from the couple small things I mentioned, this looks pretty good, and I'm super happy you got it working.

File: addons/godot-firebase/auth/auth.gd (hunk 1 above)
Comment (fenix-hub) [reply addons/godot-firebase/auth/auth.gd:488]: Honestly I wasn't aware of this method, I'll replace it!

File: addons/godot-firebase/containers/circular_container/circular_container.gd (hunk 2 above)
Comment (fenix-hub) [reply addons/godot-firebase/containers/circular_container/circular_container.gd:1]: Yep, I don't know why they are back honestly

**Review by WolfgangSenff — APPROVED** [review]

**Pull request merged**